emr/launch_emr.py

```diff
@@ -98,75 +165,57 @@ def launchemr(name=None):
             "Message", "(no message)"
         )
         print(f"Cluster Status: {state} - {message}")
-        if state in ["WAITING", "TERMINATED", "TERMINATED_WITH_ERRORS"]:
+        if state == "WAITING":
             break
+        elif state in ["TERMINATED", "TERMINATED_WITH_ERRORS"]:
+            print(f"EMR Cluster is {state}", message)
+            # print("FAIL!!! Sleeping for 2 hrs...")
+            # time.sleep(7200)
+            os.remove(cluster_id_filename)
+            raise RuntimeError(f"EMR Cluster is {state}: {message}")
         time.sleep(30)
 
     master_instance = emr.list_instances(
         ClusterId=job_flow_id, InstanceGroupTypes=["MASTER"]
     )
     ip_address = master_instance.get("Instances")[0].get("PrivateIpAddress")
-    print(f"Connecting to {ip_address}")
+    master_address = resp["Cluster"]["MasterPublicDnsName"]
+    print(f"Connecting to {master_address} at {ip_address}")
 
-    mypkey = os.path.abspath(
-        os.path.expanduser(cfg["ssh_keys"]["pkey_location"])
-    )
+    mypkey = os.path.abspath(os.path.expanduser(cfg["ssh_keys"]["pkey_location"]))
```

Comment:
Since you are calling `expanduser`, you can define `~/.ssh/dsgrid.pem` in the file.